test/arrays.coffee

```diff
@@ -70,6 +68,30 @@ test "array splats with nested arrays", ->
   list = [1, 2, a...]
   arrayEq list, [1, 2, [nonce]]
 
+test "#4260: splat after existential operator soak", ->
+  a = {b: [3]}
+  arrayEq [a?.b...], [3]
+  arrayEq [c?.b...], []
+  arrayEq [...a?.b], [3]
+  arrayEq [...c?.b], []
+
+  # Should not trigger implicit call, e.g. rest ... => rest(...)
+  arrayEq [a?.b ...], [3]
+  arrayEq [c?.b ...], []
```

Comment:
Getting a bit pedantic now, but could we add spaced prefix `...` as well? Other than that, this LGTM 👍